processing/src/main/java/io/druid/query/select/PagingSpec.java

```diff
@@ -59,22 +62,33 @@ public static PagingSpec newSpec(int threshold)
   private final Map<String, Integer> pagingIdentifiers;
   private final int threshold;
   private final boolean fromNext;
+  private final Supplier<SelectQueryConfig> configSupplier;
 
   @JsonCreator
+  @Inject
```

Comment:
You want this injected by Jackson, not a Guice injector, so remove the `@Inject` annotation and add `@JacksonInject` to the configSupplier. Check out JavaScriptExtractionFn for an example.